src/tools/networkTools.cpp

```diff
@@ -75,103 +77,144 @@ std::string kiwix::download(const std::string& url) {
+                }
+            }
+    }
+  } else {
+    std::cerr << "Call to GetAdaptersAddresses failed with error: "<< dwRetVal << std::endl;
   }
+
+  if (interfacesHead) free(interfacesHead);
+
+  return interfaces;
+}
+
 #else
-  /* Get Network interfaces information */
-  char buf[16384];
-  struct ifconf ifconf;
-  int fd = socket(PF_INET, SOCK_DGRAM, 0); /* Only IPV4 */
-  ifconf.ifc_len = sizeof(buf);
-  ifconf.ifc_buf=buf;
-  if(ioctl(fd, SIOCGIFCONF, &ifconf)!=0) {
-    perror("ioctl(SIOCGIFCONF)");
+
+std::map<std::string,ip_addr> kiwix::getNetworkInterfacesPosix() {
+  std::map<std::string,ip_addr> interfaces;
+
+  struct ifaddrs *interfacesHead;
+  if (getifaddrs(&interfacesHead) == -1) {
+      perror("getifaddrs");
   }
 
-  /* Go through each interface */
-  struct ifreq *ifreq;
-  ifreq = ifconf.ifc_req;
-  for (int i = 0; i < ifconf.ifc_len; ) {
-    if (ifreq->ifr_addr.sa_family == AF_INET) {
-      /* Get the network interface ip */
-      char host[128] = { 0 };
-      const int error = getnameinfo(&(ifreq->ifr_addr), sizeof(ifreq->ifr_addr),
-                                    host, sizeof(host),
-                                    0, 0, NI_NUMERICHOST);
-      if (!error) {
-        std::string interfaceName = std::string(ifreq->ifr_name);
-        std::string interfaceIp = std::string(host);
-        /* Add to the map */
-        interfaces[interfaceName] = interfaceIp;
-      } else {
-        perror("getnameinfo()");
+  for (ifaddrs *temp = interfacesHead; temp != NULL; temp = temp->ifa_next) {
+      if (temp->ifa_addr == NULL) continue;
+
+      if (temp->ifa_addr->sa_family == AF_INET) {
+          sockaddr_in *si = (sockaddr_in *)(temp->ifa_addr);
+          char host[INET_ADDRSTRLEN] = {0};
+          inet_ntop(AF_INET, &(si->sin_addr), host, sizeof(host));
+          interfaces[temp->ifa_name].addr=host;
+      } else if (temp->ifa_addr->sa_family == AF_INET6) {
+          sockaddr_in6 *si = (sockaddr_in6 *)(temp->ifa_addr);
+          char host[INET6_ADDRSTRLEN] = {0};
+          inet_ntop(AF_INET6, &(si->sin6_addr), host, sizeof(host));
+          if (!IN6_IS_ADDR_LINKLOCAL(&(si->sin6_addr)))
+            interfaces[temp->ifa_name].addr6=host;
       }
-    }
+  }
+
+  freeifaddrs(interfacesHead);
+
+  return interfaces;
+}
 
-    /* some systems have ifr_addr.sa_len and adjust the length that
-     * way, but not mine. weird */
-    size_t len;
-#ifndef __linux__
-    len = IFNAMSIZ + ifreq->ifr_addr.sa_len;
-#else
-    len = sizeof(*ifreq);
 #endif
-    ifreq = (struct ifreq*)((char*)ifreq+len);
-    i += len;
-  }
+
+std::map<std::string,ip_addr> kiwix::getNetworkInterfaces() {
+  std::map<std::string,ip_addr> interfaces;
+
+#ifdef _WIN32
+  return getNetworkInterfacesWin();
+#else
+  return getNetworkInterfacesPosix();
 #endif
-  return interfaces;
+
 }
 
-std::string kiwix::getBestPublicIp() {
-  auto interfaces = getNetworkInterfaces();
+std::string kiwix::getBestPublicIp(bool ipv6) {
```

Comment:
What is the meaning of the `ipv6` parameter? Does it enable (allow) returning an IPv6 address or requires it?